content/languages/shell/rspec.md

```diff
@@ -1,12 +1,37 @@
 ---
-tags:
-  - shell
-  - reference
-  - testing
+title: RSpec
+tags: [shell, reference, testing]
 ---
 
 # RSpec
 
+Shell can be tested with Ruby's [RSpec](https://rspec.info/) library which runs the solution shell script as a subprocess. The `run_shell(args: [], output: true)` function is provided to facilitate this. Calling `run_shell(2, 3, 4)` effectively runs `./solution.sh 2 3 4` and returns the script's stdout as a string.
+
+## Basic Setup
+
+### Solution
+
+```shell
+#!/bin/sh
```

Comment:
```suggestion
#!/bin/bash
```

The shebang is ignored, so this doesn't really matter, but using Bash is more correct. It's executed like `bash -c $(cat solution.sh)`.